src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/BodySubstitutionParser.cs

```diff
@@ -1,174 +1,145 @@
-                    if (!string.IsNullOrEmpty(value))
-                    {
-                        if (!TryConvertValue(value, method.ReturnType, out object? res))
+                switch (action)
+                {
+                    case "remove":
+                        _methodSubstitutions.Add(method, BodySubstitution.ThrowingBody);
+                        break;
+                    case "stub":
+                        BodySubstitution stubBody;
+                        if (method.Signature.ReturnType.IsVoid)
+                            stubBody = BodySubstitution.EmptyBody;
+                        else
+                            stubBody = BodySubstitution.Create(TryCreateSubstitution(method.Signature.ReturnType, GetAttribute(nav, "value")));
+
+                        if (stubBody != null)
                         {
-                            LogWarning(methodNav, DiagnosticId.XmlInvalidValueForStub, method.GetDisplayName());
-                            return;
+                            _methodSubstitutions[method] = stubBody;
+                        }
+                        else
+                        {
+                            // Context.LogWarning ($"Invalid value for '{method.GetDisplayName ()}' stub", 2010, _xmlDocumentLocation);
                         }
 
-                        _substitutionInfo.SetMethodStubValue(method, res);
-                    }
-
-                    _substitutionInfo.SetMethodAction(method, MethodAction.ConvertToStub);
-                    return;
-                default:
-                    LogWarning(methodNav, DiagnosticId.XmlUnkownBodyModification, action, method.GetDisplayName());
-                    return;
+                        break;
+                    default:
+                        //Context.LogWarning($"Unknown body modification '{action}' for '{method.GetDisplayName()}'", 2011, _xmlDocumentLocation);
+                        break;
+                }
             }
         }
 
-        protected override void ProcessField(TypeDefinition type, XPathNavigator fieldNav)
+        protected override void ProcessField(TypeDesc type, FieldDesc field, XPathNavigator nav)
         {
-            Debug.Assert(_substitutionInfo != null);
-            string name = GetAttribute(fieldNav, "name");
-            if (string.IsNullOrEmpty(name))
-                return;
-
-            var field = type.Fields.FirstOrDefault(f => f.Name == name);
-            if (field == null)
-            {
-                LogWarning(fieldNav, DiagnosticId.XmlCouldNotFindFieldOnType, name, type.GetDisplayName());
-                return;
-            }
-
             if (!field.IsStatic || field.IsLiteral)
             {
-                LogWarning(fieldNav, DiagnosticId.XmlSubstitutedFieldNeedsToBeStatic, field.GetDisplayName());
+                // Context.LogWarning ($"Substituted field '{field.GetDisplayName ()}' needs to be static field.", 2013, _xmlDocumentLocation);
```

Comment:
Could you leave this commented out block how it was in the IL Linker version (`DiagnosticId.XmlSubstitutedFieldNeedsToBeStatic`, etc.).